### ISSUE TITLE
Events overhaul

### ISSUE THREAD

**Issue by Guichaguri** [Guichaguri/react-native-track-player#6]
For now, events are triggered with media buttons and with only one player event (ended). It was enough for local players, as everything depends on the app. I'll create a list of categories of events to organize them:

* Media Button Events: Events that will trigger when someone clicks a button outside your app UI (such as smartwatches, notifications, etc)
* Player Events: Events that will trigger when the player state changes.
* Remote Events: Events that will trigger when remote devices were found/connected/disconnected

### Media Button Events
All of this events are not sent to the main player, so you can implement your own logic.
* `play`: When the play button is pressed
* `pause`: When the pause button is pressed
* `stop`: When the stop button is pressed
* `skipNext`: When the skip to next button is pressed
* `skipPrevious`: When the skip to previous button is pressed
* `seekTo`: When the position of the playback is changed
* `setRating`: When the rating is changed

### Player Events
* `player-state`: When the state is changed (can be used to update the UI)
* `player-load`: When a new song is loaded
* `player-play`: When the state is changed to playing
* `player-paused`: When the state is changed to paused
* `player-stop`: When the state is changed to stopped
* `player-ended`: When the playback has reached its end
* `player-error`: When an error has occurred

### Remote Events
* `device-added`: When a compatible device is found (can be used in a list)
* `device-removed`: When a compatible device can't be found anymore (can be used in a list)
* `device-connected`: When a device is connected
* `device-disconnected`: When a device is disconnected

**Comment by Guichaguri** [comment]
Player and Remote events are prefixed. I don't know if I should also prefix the media button events

**Comment by jasongrishkoff** [comment]
Prefixing media buttons might get a bit wordy.  One thing is to maybe keep the tenses the same on the player events?

player-play -> player-playing
player-stop -> player-stopped
player-load -> player-loaded
